watchdog/main.go

```diff
@@ -271,6 +299,7 @@ func main() {
 	}
 
 	http.HandleFunc("/", makeRequestHandler(&config))
+	http.HandleFunc("/_/health", makeHealthHandler())
```

Comment:
@viveksyngh how did you verify this working?

I've build the watchdog from this branch but I've been unable to hit the health endpoint?
```
$ export fprocess="echo hello"
$ ./watchdog
2018/03/17 18:16:38 Writing lock-file to: /var/folders/2z/d4szj605431693r_4pj3sfwr0000gq/T/.lock
```
Curling the function root works as expected:
```
$ curl -vvv http://localhost:8080/
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET / HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Duration-Seconds: 0.004187
< Date: Sat, 17 Mar 2018 18:17:30 GMT
< Content-Length: 6
< Content-Type: text/plain; charset=utf-8
<
hello
* Connection #0 to host localhost left intact
```

But curling the health endpoint doesn't return the expected OK, instead invoking the function again:
```
$ curl -vvv http://localhost:8080/_/health
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /_/health HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Duration-Seconds: 0.003916
< Date: Sat, 17 Mar 2018 18:18:11 GMT
< Content-Length: 6
< Content-Type: text/plain; charset=utf-8
<
hello
* Connection #0 to host localhost left intact
```
Does this actually work for you?

Reply:
Ok so your route ordering is wrong here, the health route needs to occur *before* the `/` route, currently you're never reaching the `/_/health` route as `/` is handling it.

Reply:
I verified it on my local. I can share the result. 

```
curl -i http://localhost:8081/
HTTP/1.1 200 OK
X-Duration-Seconds: 0.005071
Date: Sat, 17 Mar 2018 18:25:38 GMT
Content-Length: 1329
Content-Type: text/plain; charset=utf-8

TERM_SESSION_ID=w0t1p0:633DD086-384A-4778-B803-2F9EA6073601
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.J8jlwZ3nBI/Listeners
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.fuASYugNfM/Render
COLORFGBG=7;0
ITERM_PROFILE=Default
XPC_FLAGS=0x0
LANG=en_US.UTF-8
PWD=/Users/viveks/Repos/go-projects/src/github.com/openfaas/faas/watchdog
SHELL=/bin/zsh
TERM_PROGRAM_VERSION=3.1.5
TERM_PROGRAM=iTerm.app
PATH=/Library/Frameworks/Python.framework/Versions/3.6/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin
COLORTERM=truecolor
TERM=xterm-256color
HOME=/Users/viveks
TMPDIR=/var/folders/tw/6ww2mtms12ndnypdycqfy_q40000gy/T/
USER=viveks
XPC_SERVICE_NAME=0
LOGNAME=viveks
__CF_USER_TEXT_ENCODING=0x1FE:0:0
ITERM_SESSION_ID=w0t1p0:633DD086-384A-4778-B803-2F9EA6073601
SHLVL=1
OLDPWD=/Users/viveks/Repos/go-projects/src/github.com/openfaas/faas
ZSH=/Users/viveks/.oh-my-zsh
PAGER=less
LESS=-R
LC_CTYPE=en_US.UTF-8
LSCOLORS=Gxfxcxdxbxegedabagacad
GOPATH=/Users/viveks/Repos/go-projects/
VISUAL=vim
P4CONFIG=.perforce
P4EDITOR=/usr/bin/vim
P4PORT=rsh:ssh -2 -a -l p4source -q -x p4.source.akamai.com /bin/true
P4USER=viveks
fprocess=env
_=/Users/viveks/Repos/go-projects/src/github.com/openfaas/faas/watchdog/./watchdog
Http_User_Agent=curl/7.54.0
Http_Accept=*/*
Http_Method=GET
Http_ContentLength=0
Http_Path=/
```

```
curl -i http://localhost:8081/_/health
HTTP/1.1 200 OK
Date: Sat, 17 Mar 2018 18:25:43 GMT
Content-Length: 2
Content-Type: text/plain; charset=utf-8

OK%
```

Reply:
I just changed the port in my local because I have something else running on port `8080`.

Reply:
```
curl -i -XPOST http://localhost:8081/_/health
HTTP/1.1 405 Method Not Allowed
Date: Sat, 17 Mar 2018 18:30:29 GMT
Content-Length: 0
```

Reply:
```
2018/03/17 23:55:29 Writing lock-file to: /var/folders/tw/6ww2mtms12ndnypdycqfy_q40000gy/T/.lock
2018/03/17 23:55:38 Forking fprocess.
2018/03/17 23:55:38 Wrote 1329 Bytes - Duration: 0.005071 seconds
```

Reply:
@johnmccabe I will do that ordering.

Reply:
If you rebase with master you will find the watchdog as of 0.7.6 now allows `port` to be overridden temporarily for this kind of purpose.